tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/DeployerTest.java

```diff
@@ -0,0 +1,160 @@
+        JettyHomeTester distribution = JettyHomeTester.Builder.newInstance()
+            .jettyVersion(jettyVersion)
+            .jettyBase(jettyBase)
+            .build();
+
+        try (JettyHomeTester.Run run1 = distribution.start(List.of("--add-modules=resources,http,core-deploy")))
+        {
+            assertTrue(run1.awaitFor(START_TIMEOUT, TimeUnit.SECONDS));
+            assertEquals(0, run1.getExitValue());
+
+            Path baseResourcePath = jettyBase.resolve("work/test");
+            FS.ensureDirExists(baseResourcePath);
+            Path testXml = jettyBase.resolve("webapps").resolve("test.xml");
+            Files.writeString(testXml, """
+                <?xml version="1.0"?>
+                <!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://jetty.org/configure.dtd">
+                <Configure class="org.eclipse.jetty.server.handler.ContextHandler">
+                  <Set name="contextPath">/test</Set>
+                  <Set name="baseResourceAsPath">
+                    <Call class="java.nio.file.Path" name="of">
+                      <Arg>$R</Arg>
+                    </Call>
+                  </Set>
+                  <Set name="handler">
+                    <New class="org.eclipse.jetty.server.handler.ResourceHandler" />
+                  </Set>
+                </Configure>
+                """.replace("$R", baseResourcePath.toAbsolutePath().toString()), StandardOpenOption.CREATE);
+
+            String testFileContent = "hello";
+            Files.writeString(baseResourcePath.resolve("test.txt"), testFileContent, StandardOpenOption.CREATE);
+
+            int httpPort = Tester.freePort();
+            try (JettyHomeTester.Run run2 = distribution.start("jetty.http.port=" + httpPort))
+            {
+                assertTrue(run2.awaitConsoleLogsFor("Started oejs.Server@", START_TIMEOUT, TimeUnit.SECONDS));
+
+                startHttpClient();
+                ContentResponse response = client.GET("http://localhost:" + httpPort + "/test/test.txt");
+                assertThat(response.getStatus(), is(HttpStatus.OK_200));
+                assertThat(response.getContentAsString(), is(testFileContent));
+            }
+        }
+    }
+
+    @Test
+    public void testCoreDeployNominatedDirectoryStaticOnly() throws Exception
+    {
+        Path jettyBase = newTestJettyBaseDirectory();
+        String jettyVersion = System.getProperty("jettyVersion");
+        JettyHomeTester distribution = JettyHomeTester.Builder.newInstance()
+            .jettyVersion(jettyVersion)
+            .jettyBase(jettyBase)
+            .build();
+
+        try (JettyHomeTester.Run run1 = distribution.start(List.of("--add-modules=resources,http,core-deploy")))
+        {
+            assertTrue(run1.awaitFor(START_TIMEOUT, TimeUnit.SECONDS));
+            assertEquals(0, run1.getExitValue());
+
+            Path nominatedDir = jettyBase.resolve("webapps").resolve("test.d");
```

Comment:
"test.d" should never be deployed by the deployer. EVER!  Only if an xml sets the baseResource to a "test.d" directory should be be used as such

Reply:
It's not deployed, not even in this test.

Reply:
The intent of "/foo.d/" directories is that they are ignored by the deployer.  They should NEVER be the basis of a deployment unit, even if "/foo.d/static/" ends up being their resource base.
There is no reason we cannot have "/foo/" as the core deployment unit with "/foo/static" being the resource base.

We need to schedule a meeting to resolve #11861